HBASE-29236: Add Support for Dynamic Configuration at the Coprocessor Level

kgeisz opened the pull request:
https://issues.apache.org/jira/browse/HBASE-29236

This pull request adds support for Dynamic Configuration at the coprocessor level and makes the `hbase.global.readonly.enabled` config variable dynamically configurable.  This variable is used with the ReadOnlyController coprocessor.  In order to make this class dynamically configurable, we have it implement the ConfigurationObserver interface.  We also need to register this observer with the ConfigurationManager, so the ReadOnlyController is registered to HMaster's, HRegionServer's, and HRegion's ConfigurationManager.  Also, since the ReadOnlyController is a coprocessor, we need to use the CoprocessorHost in order to get the coprocessor and register it with the ConfigurationManager.

`hbase.global.readonly.enabled` can be configured dynamically in the following way:

1. Add `hbase.global.readonly.enabled` to hbase-site.xml:

```
<property>
  <name>hbase.global.readonly.enabled</name>
  <value>false</value>
</property>
```

2. Add the ReadOnlyController coprocessor to the master and region server by adding the following to hbase-site.xml:

```
<property>
  <name>hbase.coprocessor.master.classes</name>
  <value>org.apache.hadoop.hbase.security.access.ReadOnlyController</value>
</property>
<property>
  <name>hbase.coprocessor.regionserver.classes</name>
  <value>org.apache.hadoop.hbase.security.access.ReadOnlyController</value>
</property>
<property>
  <name>hbase.coprocessor.region.classes</name>
  <value>org.apache.hadoop.hbase.security.access.ReadOnlyController</value>
</property>
```

3. Start HBase
4. Change `hbase.global.readonly.enabled` from `false` to `true` in hbase-site.xml and save the file.
5. In the HBase shell, run `update_all_config`.
6. You should see the following log message for the master, region servers(s), and region(s):

```
2025-04-23T15:23:59,008 INFO  [RpcServer.priority.RWQ.Fifo.read.handler=3,queue=1,port=16000] access.ReadOnlyController: Config hbase.global.readonly.enabled has been dynamically changed to true
```

anmolnar commented on hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/ReadOnlyController.java:
```diff
@@ -65,26 +66,25 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(ReadOnlyController.class);
-  private Configuration conf;
+  private volatile boolean globalReadOnlyEnabled;
```
Why do we need this?
Use AtomicBoolean instead?

kgeisz replied:
I know we talked about this in our meeting.  I will add a comment here as well for completeness.  I chose to use `volatile` because that is consistent with what I have seen in other PRs for dynamic configuration.  The PRs I looked at were https://github.com/apache/hbase/pull/5823 and https://github.com/apache/hbase/pull/4181.

anmolnar replied:
👍

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m 13s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 20s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   3m 17s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  javadoc  |   3m 33s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  shadedjars  |   8m  8s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 15s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   3m  8s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 37s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 281m 46s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/1/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 325m  1s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux e3968ddf11ed 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 59de4ffc02919640903d87761a283a54be327d8f |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/1/testReport/ |
| Max. process+thread count | 6232 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 43s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  9s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |  12m 32s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   1m 36s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |  13m 50s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotless  |   1m  7s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 39s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  12m  7s |  |  the patch passed  |
| +1 :green_heart: |  javac  |  12m  7s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m 29s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |  13m 18s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  17m  5s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   1m  5s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  |  The patch does not generate ASF License warnings.  |
|  |   |  95m 28s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 6253d0dea2f0 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 59de4ffc02919640903d87761a283a54be327d8f |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 190 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/3/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  8s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   2m 51s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  javadoc  |   2m 57s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  shadedjars  |   6m 20s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 19s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 18s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 18s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 23s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 52s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 245m 42s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/3/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 282m 45s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/3/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 580ab2375bd5 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 59de4ffc02919640903d87761a283a54be327d8f |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/3/testReport/ |
| Max. process+thread count | 6084 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/3/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

anmolnar commented on hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/ReadOnlyController.java:
```diff
@@ -390,4 +390,12 @@ public void preCleanupBulkLoad(ObserverContext<RegionCoprocessorEnvironment> ctx
+  public void onConfigurationChange(Configuration conf) {
+    this.globalReadOnlyEnabled = conf.getBoolean(HConstants.HBASE_GLOBAL_READONLY_ENABLED_KEY,
+      HConstants.HBASE_GLOBAL_READONLY_ENABLED_DEFAULT);
+    LOG.info("Config {} has been dynamically changed to {}",
+      HConstants.HBASE_GLOBAL_READONLY_ENABLED_KEY, this.globalReadOnlyEnabled);
```
Shall we add a check whether the value has actually been changed here?
This info should only be logged if the previous value was different.

kgeisz replied:
Sounds good.  I'll add a check here.

anmolnar reviewed (changes requested):
Patch looks good to me. Thanks @kgeisz !
I think this is the point that you should work together with @sharmaar12 and add unit test which starts up a non-RO mini cluster, dynamically changes it to RO mode and validate some mutatation commands.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 34s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 36s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 30s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   8m  9s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   1m 12s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |   9m  3s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m  2s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   8m  2s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m 12s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   9m 19s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 54s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 19s |  |  The patch does not generate ASF License warnings.  |
|  |   |  66m  8s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 66d119c97973 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 511c8888ac89f4929c2b860e5a82309098ffd76a |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 189 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/4/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Kota-SH commented on hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/CoprocessorHost.java:
```diff
@@ -116,6 +118,17 @@ public Set<String> getCoprocessors() {
     return returnValue;
   }
 
+  public void registerConfigurationObservers(ConfigurationManager configurationManager) {
```
nit: Can you add a Javadoc for this method, to be consistent with other methods in this class?

kgeisz replied:
Sure, will do.

Kota-SH reviewed (commented):
LGTM, just a couple of nits.
+1 for the unit tests.

kgeisz commented:
@anmolnar, I have added some unit test cases.  I also did a couple more things:

1. I had to register the ReadOnlyController to HRegion's ConfigurationManager in order for an exception to get thrown when a Put command is performed with `hbase.global.readonly.enabled` set to `false`.
2. I removed the WALEdit checks inside of `prePut()` and `preBatchMutate()` for ReadOnlyController.java.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 45s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 56s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   9m 30s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   1m 18s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |  10m 27s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotless  |   0m 47s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  7s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  10m 25s |  |  the patch passed  |
| +1 :green_heart: |  javac  |  10m 25s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   9m 22s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 45s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| -1 :x: |  spotless  |   0m 44s |  |  patch has 1 errors when running spotless:check, run spotless:apply to fix.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 19s |  |  The patch does not generate ASF License warnings.  |
|  |   |  72m 49s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/6/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 6f2146bc359b 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 5fa6a0b16abbe93161b948fdaa500fc2348a9d81 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/6/artifact/yetus-general-check/output/patch-spotless.txt |
| Max. process+thread count | 189 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/6/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 31s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 10s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   2m  6s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  javadoc  |   2m 21s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  shadedjars  |   5m 55s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  7s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  9s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  9s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 22s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 51s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 260m 28s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/6/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 294m  0s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/6/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux ab7e266aeec8 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 5fa6a0b16abbe93161b948fdaa500fc2348a9d81 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/6/testReport/ |
| Max. process+thread count | 5946 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/6/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 30s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 13s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   7m 54s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   1m  9s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |   8m 48s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  2s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 46s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   7m 46s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m 13s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   9m  2s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 46s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 20s |  |  The patch does not generate ASF License warnings.  |
|  |   |  63m 56s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/7/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 4c70bcefdaea 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 366c342bb624919f756bbc0da74942cb521543b3 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 192 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/7/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 44s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 32s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   2m  6s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  javadoc  |   2m 20s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  shadedjars  |   5m 53s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  1s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  9s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  9s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 26s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 48s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 258m 23s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/7/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 292m 42s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/7/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux ccebdc66579d 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 366c342bb624919f756bbc0da74942cb521543b3 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/7/testReport/ |
| Max. process+thread count | 6014 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/7/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 30s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 24s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   7m 50s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   1m 15s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |   8m 54s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 57s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 50s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   7m 50s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m 13s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   9m 18s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 49s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 20s |  |  The patch does not generate ASF License warnings.  |
|  |   |  65m  6s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/8/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux cb56549c5ef8 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 5e2854e46ad86731e84a45ee4dc1f3d9dfe55fde |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 190 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/8/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 31s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 14s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   2m 11s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  javadoc  |   2m 26s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  shadedjars  |   6m  2s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 11s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 11s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 23s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 58s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 248m 52s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/8/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 282m 52s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/8/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux b8a00d628720 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 5e2854e46ad86731e84a45ee4dc1f3d9dfe55fde |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/8/testReport/ |
| Max. process+thread count | 6141 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/8/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 32s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 12s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   2m 14s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  javadoc  |   2m 24s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  shadedjars  |   5m 58s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 16s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 16s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 24s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 57s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   3m 11s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/9/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   |  33m 48s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/9/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 78ff91c15e74 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 5e2854e46ad86731e84a45ee4dc1f3d9dfe55fde |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/9/testReport/ |
| Max. process+thread count | 312 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/9/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

anmolnar commented on hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/ReadOnlyController.java:
```diff
@@ -100,9 +99,6 @@ public Optional<RegionObserver> getRegionObserver() {
   @Override
   public void prePut(ObserverContext<? extends RegionCoprocessorEnvironment> c, Put put,
     WALEdit edit) throws IOException {
-    if (edit.isMetaEdit() || edit.isEmpty()) {
-      return;
-    }
```
So, you don't need these checks anymore, because you start the mini cluster in R/W mode in your tests. Does that mean a real cluster can be started in R/O mode, have you verified that?

kgeisz replied:
@anmolnar, I cannot start HBase in Read-Only mode with these current changes.

anmolnar reviewed (approved):
lgtm. Thanks for the tests!

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 30s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  9s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   7m 49s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   1m 12s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |   8m 49s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 58s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   8m  3s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   8m  3s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m  9s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   9m  6s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 45s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 20s |  |  The patch does not generate ASF License warnings.  |
|  |   |  66m  4s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/9/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux e53fb6d22ae9 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 5e2854e46ad86731e84a45ee4dc1f3d9dfe55fde |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 192 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/9/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  15m  6s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  1s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m  3s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |  13m 12s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   4m  5s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |  20m 35s |  |  HBASE-29081 passed  |
| -1 :x: |  spotless  |   1m 21s |  |  branch has 1 errors when running spotless:check, run spotless:apply to fix.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   8m 51s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  10m 32s |  |  the patch passed  |
| +1 :green_heart: |  javac  |  10m 33s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m 40s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |  18m 39s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  23m 55s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   1m 20s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 29s |  |  The patch does not generate ASF License warnings.  |
|  |   | 143m 22s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/10/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux fe720c84fc68 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / 035cfdb441b96a43217a95f93323f421b354b763 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/10/artifact/yetus-general-check/output/branch-spotless.txt |
| Max. process+thread count | 190 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/10/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   9m 35s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  2s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  2s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   9m 33s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  26m 50s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |  10m 29s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   1m 45s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |  27m  3s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotless  |   1m 41s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   8m 50s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  10m 18s |  |  the patch passed  |
| +1 :green_heart: |  javac  |  10m 18s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   1m 28s | [/results-checkstyle-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/11/artifact/yetus-general-check/output/results-checkstyle-root.txt) |  root: The patch generated 3 new + 29 unchanged - 0 fixed = 32 total (was 29)  |
| +1 :green_heart: |  spotbugs  |  21m 54s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  23m 16s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| -1 :x: |  spotless  |   0m 57s |  |  patch has 1 errors when running spotless:check, run spotless:apply to fix.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 41s |  |  The patch does not generate ASF License warnings.  |
|  |   | 172m 42s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/11/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 89054daa2a5f 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / faf86e313cc5664e29ff504ec3732ff1131f60dd |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/11/artifact/yetus-general-check/output/patch-spotless.txt |
| Max. process+thread count | 188 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/11/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

NihalJain commented on hbase-server/src/test/java/org/apache/hadoop/hbase/security/access/TestReadOnlyController.java:
```diff
@@ -69,20 +80,39 @@ public class TestReadOnlyController {
+      TEST_UTIL.deleteTable(TEST_TABLE);
+      connection.close();
+      TEST_UTIL.shutdownMiniCluster();
+      throw new RuntimeException(e);
```
I hope this would not cause the whole test pipeline to fail?

NihalJain commented on hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java:
```diff
@@ -609,6 +609,12 @@ protected String getUseThisHostnameInstead(Configuration conf) {
   private void registerConfigurationObservers() {
     configurationManager.registerObserver(this.rpcServices);
     configurationManager.registerObserver(this);
+    if (cpHost != null) {
```
Do we not need to deregister in case of Master also?

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m 40s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 41s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   2m 12s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  javadoc  |   2m 25s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  shadedjars  |   6m  2s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 12s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 12s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 24s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 20s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 342m 24s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/12/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 401m  8s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/12/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux daebca6d119c 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / faf86e313cc5664e29ff504ec3732ff1131f60dd |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/12/testReport/ |
| Max. process+thread count | 5130 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/12/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

kgeisz commented:
@NihalJain, thanks for the feedback!  I updated the name of the Jira ticket and added some info to the ticket's description.  I also updated the name of the PR to match the Jira ticket.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 31s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ HBASE-29081 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 17s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  compile  |   7m 53s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  checkstyle  |   1m 10s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotbugs  |   8m 58s |  |  HBASE-29081 passed  |
| +1 :green_heart: |  spotless  |   0m 47s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  6s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 52s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   7m 52s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   1m 12s | [/results-checkstyle-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/13/artifact/yetus-general-check/output/results-checkstyle-root.txt) |  root: The patch generated 3 new + 29 unchanged - 0 fixed = 32 total (was 29)  |
| +1 :green_heart: |  spotbugs  |   9m  7s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 58s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 20s |  |  The patch does not generate ASF License warnings.  |
|  |   |  65m  4s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/13/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6931 |
| JIRA Issue | HBASE-29236 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 8044e1bfec8e 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-29081 / faf86e313cc5664e29ff504ec3732ff1131f60dd |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 190 (vs. ulimit of 30000) |
| modules | C: hbase-server . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6931/13/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

anmolnar commented:
Build timed out, but it was successful, so I merge the patch regardless. Thanks @kgeisz !
